src/generators/essudpgen/ReadoutGeneratorBase.cpp

```diff
@@ -48,6 +48,20 @@ ReadoutGeneratorBase::ReadoutGeneratorBase(Parser::DetectorType Type) {
   app.add_flag("-l, --loop", Settings.Loop, "Run forever");
 
   Settings.Type = Type;
```

Comment:
Since we can change the type from the configuration, I would suggest to make the DetectorType a more inteligent "enum" see my other comment about this. Then we can remove it to be part of the base class initial parameter and just add as an option when generator starts.

Reply:
Potential for future improvement